### PR TITLE
feat: introduce setup-jrsonnet action

### PR DESCRIPTION
**Pull request by Duologic**
I'm using a version of this action in https://github.com/grafana/grafana-crossplane-libsonnet but think it would be good to maintain this as a shared workflow.

### CODE REVIEW

**Comment by zerok** [comment]
Would this perhaps even make sense as a separate repository? I imagine for instance that other Tanka users might be an audience here but not of the rest of the shared workflows.

**Comment by Duologic** [comment]
How do we decide what should or shouldn't go into this repo?

**Comment by zerok** [comment]
We have a broad scope defined here: https://github.com/grafana/shared-workflows/blob/main/CONTRIBUTING.md#scope-of-shared-workflows

I guess it comes down to what other repos would need this.

**Comment by iainlane** [comment]
These `setup-` actions would be nice contributions to the upstream projects themselves too.

It doesn't necessarily need to block us adding them in here while that process is ongoing, though - that's one for folks' best judgement. 🙂

(Just making a comment/suggestion, not weighing in on the review - please carry on reviewing/discussing.)

File: actions/setup-jrsonnet/README.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Setup Jrsonnet
+
+Setup Jrsonnet cli and add it to the PATH, this action will pull the binary from GitHub releases and store it in cache for the next run.
```
Comment (zerok) [inline actions/setup-jrsonnet/README.md:3]:
```suggestion
# Setup jrsonnet

Setup jrsonnet CLI and add it to the PATH, this action will pull the binary from GitHub releases and store it in cache for the next run.
```

File: actions/setup-jrsonnet/action.yaml [hunk 2]
```diff
@@ -0,0 +1,64 @@
+name: Setup Jrsonnet
+description: Setup Jrsonnet cli and add it to the PATH, this action will pull the binary from GitHub releases and store it in cache for the next run.
```
Comment (zerok) [inline actions/setup-jrsonnet/action.yaml:2]:
```suggestion
name: Setup jrsonnet
description: Setup jrsonnet CLI and add it to the PATH, this action will pull the binary from GitHub releases and store it in cache for the next run.
```

File: actions/setup-jrsonnet/action.yaml [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        key: ${{ inputs.cache-prefix }}-${{ runner.os }}-${{ runner.arch }}-${{ inputs.version }}
+
+    - name: Map OS and ARCH to Jrsonnet release artifact
+      if: steps.cache.outputs.cache-hit != 'true'
+      shell: sh
+      run: |
+        echo "OS=$(go env GOOS)" >> "$GITHUB_ENV"
+        echo "ARCH=$(go env GOARCH)" >> "$GITHUB_ENV"
+
+    - name: Fetch Github Release Asset
```
Comment (zerok) [inline actions/setup-jrsonnet/action.yaml:37]:
```suggestion
    - name: Fetch GitHub Release Asset
```

File: actions/setup-jrsonnet/action.yaml [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+  steps:
+    - name: Setup cache
+      id: cache
+      uses: actions/cache@6849a6489940f00c2f30c0fb92c6274307ccb58a # v4.1.2
+      with:
+        path: ${{ github.workspace }}/bin/jrsonnet
+        key: ${{ inputs.cache-prefix }}-${{ runner.os }}-${{ runner.arch }}-${{ inputs.version }}
+
+    - name: Map OS and ARCH to Jrsonnet release artifact
```
Comment (zerok) [inline actions/setup-jrsonnet/action.yaml:30]:
```suggestion
    - name: Map OS and ARCH to jrsonnet release artifact
```

**Review by zerok — APPROVED** [review]
LGTM 🙂 Just some minor suggestions for consistency.

**Comment by zerok** [comment]
@Duologic What do you think about, as Iain suggested, making this action available as part of the upstream repository? https://github.com/CertainLach/jrsonnet 🙂

**Comment by Duologic** [comment]
I'm not opposed but like that we are currently in control of it, handing it over would require more back and forth in case we need to change it. Is it okay to wait with this for like a month so that get some mileage on this?

**Comment by zerok** [comment]
Absolutely 🙂

**Pull request merged**